bindings/pyroot/cppyy/CPyCppyy/src/API.cxx

```diff
@@ -413,7 +413,21 @@ void CPyCppyy::ExecScript(const std::string& name, const std::vector<std::string
                           << std::endl;
             }
          };
-         WideStringListAppendHelper(&fConfig.argv, Py_GetProgramName());
+
+         // The first element of the PyWideStringList can be arbitray, which
+         // you can seen by running the following:
+         //
+         // Content of test.py
+         // ```python
+         // print(sys.executable) # will print the executable, e.g. "python" or "root"
+         // print(sys.argv)       # will print ['test.py', '1', '2', '3'] in this example
+         // ```
+         //
+         // ```c++
+         // CPyCppyy::ExecScript("test.py", sizeof(argv)/sizeof(argv[0]), argv);
+         // ```
+         WideStringListAppendHelper(&fConfig.argv, std::wstring{}.c_str());
```

Comment:
Although I trust this is what happens, by reading the docs at https://docs.python.org/3/c-api/init.html#c.Py_GetProgramName  it looks to me that the more appropriate string to pass would be `fConfig.program_name` member

Reply:
Oh that's a pretty good idea!

Reply:
Actually it doesn't work, because we never set `fConfig.program_name`. So it's a `nullptr`, and then there is a segfault down the line.